sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_event.hpp

```diff
@@ -111,6 +116,45 @@ struct _ur_ze_event_list_t {
     }
     return *this;
   }
+
+  // This function allows to merge two _ur_ze_event_lists
+  // The ur_ze_event_list "other" is added to the caller list.
+  // Note that new containers are allocated to contains the additional elements.
+  void fusion(const _ur_ze_event_list_t &other) {
+    if (this != &other) {
+      // save of the previous object values
+      uint32_t preLength = this->Length;
+      ze_event_handle_t *preZeEventList = this->ZeEventList;
+      ur_event_handle_t *preUrEventList = this->UrEventList;
+
+      // allocate new memory
+      uint32_t Length = preLength + other.Length;
+      this->ZeEventList = new ze_event_handle_t[Length];
+      ;
+      this->UrEventList = new ur_event_handle_t[Length];
+
+      // copy elements
+      uint32_t TmpListLength = 0;
+      uint32_t I;
+      for (I = 0; I < preLength; I++) {
+        this->ZeEventList[TmpListLength] = preZeEventList[I];
+        this->UrEventList[TmpListLength] = preUrEventList[I];
+        TmpListLength += 1;
+      }
+      for (I = 0; I < other.Length; I++) {
+        this->ZeEventList[TmpListLength] = other.ZeEventList[I];
+        this->UrEventList[TmpListLength] = other.UrEventList[I];
```

Comment:
Do we need to be retaining the copied events from `other` here? Since that is the behaviour when event lists are created, these events will now exist in two lists but only be retained once in the original list.

Reply:
In practice, I think this is no a main concern, events are released with an explicit call to the release function and are not released automatically when a list is deleted.
However, I agree that we do not need to copy these events. I therefore opted for move semantic in the new commit.